src/applications/CMakeLists.txt

```diff
@@ -430,9 +430,31 @@ FOREACH(subdir ${SUBDIRECTORIES})
       SUBDIRLIST(SUBDIRECTORIES ${SUBDIRPATH}/lib)
       # MESSAGE("[DEBUG] ItkSnap subdirs:${SUBDIRECTORIES}")
       FOREACH(itkSnapLib ${SUBDIRECTORIES})
-        INSTALL(DIRECTORY ${SUBDIRPATH}/lib/${itkSnapLib}
-          DESTINATION lib
+
+        SUBDIRLIST(SUBDIRECTORIES_SNAP ${SUBDIRPATH}/lib/${itkSnapLib})
+
+        # install all directories inside
+        FOREACH(itkSnapLib_subDirs ${SUBDIRECTORIES_SNAP})
+          INSTALL(DIRECTORY ${SUBDIRPATH}/lib/${itkSnapLib}/${itkSnapLib_subDirs}
+           DESTINATION lib/${itkSnapLib}
+           DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
+          )
+        ENDFOREACH()
+        
+        # find stand-alone files and install them separately
+        FILE( GLOB FILES_SNAP
+          LIST_DIRECTORIES FALSE
+          "${SUBDIRPATH}/lib/${itkSnapLib}/*"
         )
+
+        FOREACH(itkSnapLib_files ${FILES_SNAP})
+          INSTALL(
+            FILES ${itkSnapLib_files}
+            DESTINATION lib/${itkSnapLib}
+            PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
+          )
+        ENDFOREACH()
+
```

Comment:
Wouldn't it be easier to just make the one file (entry point) of itk-snap executable? Not everything has to be given the permission, especially directories which executable means a whole different thing.

Reply:
Oh, I just realized what you are doing. I guess this is easier; sorry